### ISSUE TITLE
HMR doesn't work for WebExtension content script

### ISSUE THREAD

**Issue by LunNova** [parcel-bundler/parcel#7164]
# 🐛 bug report

`HMR_PORT` is left as `null` for content scripts even when explicitly set with `--hmr-port`.

This is because there is logic which does not set this if it matches the web port, but for content scripts they run in the context of the page and it must also be explicitly set.

https://github.com/parcel-bundler/parcel/blob/87a44b97668666149807507aa73e3f2231ade0ab/packages/runtimes/hmr/src/HMRRuntime.js#L29-L36

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "devDependencies": {
    "@parcel/config-webextension": "^2.0.0",
    "parcel": "^2.0.0"
  },
  "targets": {
    "webext-dev": {
      "sourceMap": {
        "inline": true,
        "inlineSources": true
      },
    },
    "webext-prod": {}
  },
  "scripts": {
    "start": "parcel src/manifest.json --host localhost --port 1234 --hmr-port 1234 --target webext-dev --config @parcel/config-webextension",
    "build": "parcel build src/manifest.json --target webext-prod --config @parcel/config-webextension"
  }
}

```

## 🤔 Expected Behavior

HMR should connect successfully and work in a content script.

## 😯 Current Behavior

HMR does not connect. Error doesn't have a message set.

![image](https://user-images.githubusercontent.com/782440/138540911-a8483be0-8552-4807-bd7f-03fc9b9592b3.png)

![image](https://user-images.githubusercontent.com/782440/138540937-cb2988e6-2064-4934-9705-7df87eb1d314.png)

## 💁 Possible Solution

https://github.com/parcel-bundler/parcel/blob/87a44b97668666149807507aa73e3f2231ade0ab/packages/runtimes/hmr/src/HMRRuntime.js#L29-L36

I think to fix this the `HMR_PORT` should never be left null if `HMR_HOST` is also set. Alternatively, the port could be overridden if `HMR_HOST` is set and does not match the current `window.location`.

## 🔦 Context

I am trying out writing a WebExtension and using parcel. Have never used parcel before.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0
| Node             |  v16.11.1
| npm/Yarn         |  8.0.0
| Operating System | Windows 11

**Comment by LunNova** [comment]
Changing it back to `var HMR_PORT = ${JSON.stringify(port != null ? port : null)};` in `HMRRuntime.js` works locally for me.

**Comment by 101arrowz** [comment]
We're updating the docs to recommend `parcel watch` instead which fixes the issue. Refer to https://github.com/parcel-bundler/website/pull/1014.

**Comment by 101arrowz** [comment]
I'd say this issue can be closed as the docs have been updated to recommend `parcel watch` over `parcel serve` in development mode. Again, it's impossible to fix this without reverting #6005 as mentioned in #6061, which would break HMR over HTTP proxies like Ngrok.
